service/ui.go

```diff
@@ -358,6 +358,7 @@ func (uis *UIServer) GetServiceApp() *gimlet.APIApp {
 		FindTarget:        uis.getHostDNS,
 		StripSourcePrefix: true,
 		RemoteScheme:      "http",
+		ErrorHandler:      uis.handleBackendError("IDE service is not available", http.StatusInternalServerError),
```

Comment:
is there some call to action that we can give users to take to remedy this situation? 

can we put retry logic in the proxy for blips/restarts of up-to 10s?

Reply:
I added a line to the response to check the code-server service is running.

I added a custom Transport to the ReverseProxy (in evergreen-ci/gimlet#78) and passed in a rehttp RoundTripper.